R/calculateUnifrac.R

```diff
@@ -126,43 +126,41 @@ setMethod("calculateUnifrac",
                     tree = "missing"),
     function(x, assay.type = assay_name, assay_name = exprs_values, exprs_values = "counts", 
             tree.name = tree_name, tree_name = "phylo", transposed = FALSE, ...){
+        # Get functions and parameters based on direction
+        tree_present_FUN <- switch(as.character(transposed), "TRUE" = .check_colTree_present, 
+            "FALSE" = .check_rowTree_present, stop("."))
+        tree_FUN <- switch(as.character(transposed), "TRUE" = colTree, "FALSE" = rowTree, stop("."))
+        links_FUN <- switch(as.character(transposed), "TRUE" = colLinks, "FALSE" = rowLinks, stop("."))
+        subsetting_FUN <- switch(as.character(transposed), "TRUE" = function(x, idx) x[, idx], 
+            "FALSE" = function(x, idx) x[idx, ], stop("."))
+        transpose_MAT <- switch(as.character(transposed), "FALSE" = function(mat) t(mat), 
+            "TRUE" = function(mat) mat, stop("."))
```

Comment:
These are more clear to keep in the code as

```
if(transpose){
    mat <- t(mat)
}
```

The idea is to make the code easier to read

Reply:
Also add input check for transpose

And this might be simpler

var <- FALSE
FUN <- switch(as.numeric(var) + 1, cat, print)

Reply:
Also remove extra line breaks

Reply:
I applied these changes